### ISSUE TITLE
How do you cross-compile this lib?

### ISSUE THREAD

**Issue by xeoncross** [targodan/ffgopeg#10]
I am using ffgopeg in a project and building natively on mac works

    $ go build *.go

But trying to cross-compile does not:

    env GOOS=windows GOARCH=x64 go build *.go            
    > ffgopeg/media.go:13:2: no buildable Go source files in /....../gopkg.in/targodan/ffgopeg.v1/avformat

I am new to cross-compiling and can only assume that the [required libavfilter c libs](https://github.com/targodan/ffgopeg/blob/master/avfilter/avfilter.go#L11) must include versions for the target OS in order for this to work.

What do I need to do?

**Comment by targodan** [comment]
I have never actually thought about cross compiling this. :thinking: 

I know that without native libraries cross compiling in go is absolutely painless. Just specify 2 environment variables `GOOS` and `GOARCH` and run `go build`. But I don't know about cross compiling with native libraries.

I would imagine you definitely need the cross compiled libraries but from there I don't know.

Maybe someone with experience in this can help.

**Comment by targodan** [comment]
Ok a quick google later we are slightly wiser: https://github.com/karalabe/xgo
That project seems to do exactly what we need. You could try this out. The dependency would of course be ffmpeg, so e.g. this link: http://ffmpeg.org/releases/ffmpeg-3.3.1.tar.bz2

So the required command line would likely be `xgo --deps=http://ffmpeg.org/releases/ffmpeg-3.3.1.tar.bz2  --targets=windows/* github.com/targodan/ffgopeg`

Let me know if you succeed. :smile:

**Comment by targodan** [comment]
@Xeoncross Had any luck?

**Comment by xeoncross** [comment]
Sorry for the delay, I'll be testing this soon. I did however notice that I have to install FFmpeg on the target system as well which perhaps xgo could help with.

```
Crashed Thread:        0

Exception Type:        EXC_BREAKPOINT (SIGTRAP)
Exception Codes:       0x0000000000000002, 0x0000000000000000
Exception Note:        EXC_CORPSE_NOTIFY

Application Specific Information:
dyld: launch, loading dependent libraries

Dyld Error Message:
  Library not loaded: /usr/local/opt/ffmpeg/lib/libavutil.55.dylib
  Referenced from: /Applications/Hello.app/Contents/MacOS/Hello
  Reason: image not found

Binary Images:
    0x7fff6e592000 -     0x7fff6e5c9a47  dyld (360.22) <884763FC-CC0F-31CC-ACC4-75A805CE401D> /usr/lib/dyld
    0x7fff877c1000 -     0x7fff87b15fff  com.apple.Foundation (6.9 - 1259.22) <C83D734C-6885-3549-80FE-70DEB04318D5> /System/Library/Frameworks/Foundation.framework/Versions/C/Foundation
    0x7fff950c5000 -     0x7fff95ceeff7  com.apple.AppKit (6.9 - 1404.47) <F3411F6E-DD87-34D0-8C68-C69B2205E41D> /System/Library/Frameworks/AppKit.framework/Versions/C/AppKit
    0x7fff96b8a000 -     0x7fff96b8bffb  libSystem.B.dylib (1226.10.1) <8B780EC7-53B7-3844-B3D8-5E81F5CA598C> /usr/lib/libSystem.B.dylib
```

**Comment by targodan** [comment]
No, sweat. I just wanted to prevent this issue from being open indefinitely. :wink:

Yes, since ffmpeg (at least usually) is a dynamic and not a static library you propably need to install it on the target as well. Mayyyyyybe xgo has a way of compiling it into that, I doubt it but I don't know.
